cumulusci/tasks/connectedapp.py

```diff
@@ -0,0 +1,174 @@
+    task_options = {
+        "label": {
+            "description": "The label for the connected app.  Must contain only alphanumeric and underscores",
+            "required": True,
+        },
+        "email": {
+            "description": "The email address to associate with the connected app.  Defaults to email address from the github service if configured.",
+            "required": True,
+        },
+        "username": {
+            "description": "Create the connected app in a different org.  Defaults to the defaultdevhubusername configured in sfdx.",
+            "required": False,
+        },
+        "connect": {
+            "description": "If True, the created connected app will be stored as the CumulusCI connected_app service in the keychain.",
+            "required": False,
+        },
+        "overwrite": {
+            "description": "If True, any existing connected_app service in the CumulusCI keychain will be overwritten.  Has no effect if the connect option is False.",
+            "required": False,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        self.client_id = None
+        self.client_secret = None
+        kwargs["command"] = "force:mdapi:deploy --wait 5"
+        super(CreateConnectedApp, self)._init_options(kwargs)
+
+        # Validate label
+        if not re.match(r"^\w+$", self.options["label"]):
+            raise TaskOptionsError(
+                "label value must contain only alphanumeric or underscore characters"
+            )
+
+        # Default email to the github service's email if configured
+        if "email" not in self.options:
+            try:
+                github = self.project_config.keychain.get_service("github")
+            except ServiceNotConfigured:
+                raise TaskOptionsError(
+                    "Could not determine a default for option 'email'.  Either configure the github service using 'cci service connect github' or provide a value for the 'email' option"
+                )
+            self.options["email"] = github.email
+
+        # Default to sfdx defaultdevhubusername
+        if "username" not in self.options:
+            self._set_default_username()
+        self.options["command"] += " -u {}".format(self.options.get("username"))
+
+        self.options["connect"] = process_bool_arg(self.options.get("connect", False))
+        self.options["overwrite"] = process_bool_arg(
+            self.options.get("overwrite", False)
+        )
+
+    def _set_default_username(self):
+        self.logger.info("Getting username for the default devhub from sfdx")
+        self._run_command(
+            command="{} force:config:get defaultdevhubusername --json".format(SFDX_CLI),
+            env=self._get_env(),
+            output_handler=self._process_devhub_output,
+        )
+
+    def _process_json_output(self, line):
+        try:
+            data = json.loads(line)
+            return data
+        except Exception:
```

Comment:
ValueError should be specific enough

Reply:
Nope, in python 3 it's a JSONDecodeError

Reply:
Well, I guess I should check if JSONDecodeError subclasses ValueError :)

Reply:
>>> from json.decoder import JSONDecodeError
>>> isinstance(JSONDecodeError, ValueError)
False

Reply:
yea, because the object `JSONDecodeError` is not an instance of `ValueError`, its a subclass, so you'd need to check if an INSTANCE of jsondecodeerror isinstance of valuerror...

```
In [12]: try: json.loads('***')
    ...: except ValueError: print('hello')

Out [12]: hello
```

Reply:
What I'm really after here is to catch anything, log a message that there was an issue, and then raise the exception that was thrown.  It seems better to cast a broad net in that use case.